src/providers/ruby.rs

```diff
@@ -0,0 +1,119 @@
+                    let node_pkg = NodeProvider::get_nix_node_pkg(&package_json, env)?;
+                    if NodeProvider::get_package_manager(app)? == "pnpm" {
+                        let mut pnpm_pkg = Pkg::new("nodePackages.pnpm");
+                        // Only override the node package if not the default one
+                        if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
+                            pnpm_pkg = pnpm_pkg.set_override("nodejs", node_pkg.name.as_str());
+                        }
+                        packages.push(pnpm_pkg);
+                    } else if NodeProvider::get_package_manager(app)? == "yarn" {
+                        let mut yarn_pkg = Pkg::new("yarn");
+                        // Only override the node package if not the default one
+                        if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
+                            yarn_pkg = yarn_pkg.set_override("nodejs", node_pkg.name.as_str());
+                        }
+                        packages.push(yarn_pkg)
+                    }
+                }
+                Ok(Some(SetupPhase::new(packages)))
+            }
+            Framework::Vanilla => Ok(Some(SetupPhase::new(packages))),
+        }
+    }
+
+    fn install(&self, app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
+        let mut install_cmd = Vec::<&str>::new();
+        if NodeProvider::get_package_manager(app)? == "pnpm" {
+            install_cmd.push("pnpm i --frozen-lockfile");
+        } else if NodeProvider::get_package_manager(app)? == "yarn" {
+            if app.includes_file(".yarnrc.yml") {
+                install_cmd
+                    .push("yarn set version berry && yarn install --immutable --check-cache");
+            } else {
+                install_cmd.push("yarn install --frozen-lockfile --production=false");
+            }
+        } else if app.includes_file("package-lock.json") {
+            install_cmd.push("npm ci");
+        }
+        if app.includes_file("Gemfile") {
+            install_cmd.push("bundle install --frozen");
+        }
+
+        if install_cmd.is_empty() {
+            Ok(None)
+        } else {
+            Ok(Some(InstallPhase::new(install_cmd.join(" && "))))
+        }
+    }
+
+    fn build(&self, _app: &App, _env: &Environment) -> Result<Option<BuildPhase>> {
+        Ok(None)
+    }
+
+    fn start(&self, app: &App, _env: &Environment) -> Result<Option<StartPhase>> {
+        if app.includes_file("main.rb") {
+            return Ok(Some(StartPhase::new(String::from(
+                "bundle exec ruby main.rb",
+            ))));
+        }
+        Ok(None)
+    }
+
+    fn environment_variables(
+        &self,
+        _app: &App,
+        _env: &Environment,
+    ) -> Result<Option<EnvironmentVariables>> {
+        Ok(None)
+    }
+}
+
+impl RubyProvider {
+    fn detect_framework(&self, app: &App) -> Framework {
+        if app.includes_file("Rakefile") {
```

Comment:
I don't think the presence of a `Rakefile` indicates a Rails app. I work on projects using the Bridgetown framework, Roda, etc. and they have Rakefiles. Probably need to look for a Rails initializer or something like that.

Reply:
Are there any files that are specific to Rails?

Reply:
Should be able to just check the Gemfile for `gem 'rails'`

Reply:
I think I did a similar thing for Django

Reply:
Cause also pull the ruby version out of it if you haven't already

https://github.com/railwayapp/starters/blob/master/examples/rails-blog/Gemfile#L4

Reply:
Ok, I plan to detect all frameworks by extracting Gems. I'll also add support for different Ruby versions.